### ISSUE TITLE
How to minimize a score obtained from the prediction of a model in keras

### ISSUE THREAD

**Issue by amaall** [keras-team/keras#8434]
suppose i have  the following:
```
inputs1= Input(shape=(15,))
inputs2= Input(shape=(1,))
x = Dense(15,activation='relu')(inputs)
xp = keras.layers.concatenate([x,input2])
xp = Dense(16,activation='relu')(xp)
pred = Dense(1)(xp)
clas = Dense(1,activation='sigmoid')(x)
model = Model(inputs=[inputs1,inputs2],outputs=[pred,clas])
model.compile(loss={'pred':'mean_squared_error', 'classic':'binary_crossentropy'}, optimizer='adam')
"""  where data1,data2, ypred,and ylabel are supplied from my data file """
model.fit({'inputs1':data1,'inputs2':data2},{'pred':ypred,'clas':ylabel},epochs=50, batch_size=100)
pred_outcome = model.predict([inputs1,inputs2])
score = mean_square_error ( get_pred(a1,a2,pred_outcome) , get_true() )

```
""" My problem is that I want to minimize the mse(or any better loss that I could use in this context) of the score. Both the 2 functions produce an array of the same length, and note that the first function in the score uses the pred_outcome produced by the model. Or is their any smarter way of achieving  this while jointly learning pred and clas?"""
